src/admin/__tests__/deleteGroups.spec.js

```diff
@@ -0,0 +1,79 @@
+const createAdmin = require('../index')
+const createConsumer = require('../../consumer')
+const createProducer = require('../../producer')
+
+const {
+  createCluster,
+  newLogger,
+  createTopic,
+  secureRandom,
+  createModPartitioner,
+  waitForConsumerToJoinGroup,
+  waitForMessages,
+} = require('testHelpers')
+
+describe('Admin', () => {
+  let admin, topicName, groupId, cluster, consumer, producer
+
+  beforeEach(async () => {
+    topicName = `test-topic-${secureRandom()}`
+    groupId = `consumer-group-id-${secureRandom()}`
+
+    cluster = createCluster()
+    admin = createAdmin({ cluster: cluster, logger: newLogger() })
+    consumer = createConsumer({
+      cluster,
+      groupId,
+      maxWaitTimeInMs: 100,
+      logger: newLogger(),
+    })
+
+    producer = createProducer({
+      cluster,
+      createPartitioner: createModPartitioner,
+      logger: newLogger(),
+    })
+
+    await Promise.all([admin.connect(), consumer.connect(), producer.connect()])
+  })
+
+  afterEach(async () => {
+    // Checking that they exist first, in case the test is skipped or failed before instantiating the admin/consumer
+    admin && (await admin.disconnect())
+    consumer && (await consumer.disconnect())
+    producer && (await producer.disconnect())
+  })
+
+  describe('deleteGroups', () => {
+    test('delete groups', async () => {
```

Comment:
Yes, definitely, I found out now that if you delete group if it is still connected it gives error, can you tell me what would be the best way to represent response? Is it ok to do i like this:
```
{ throttleTimeMs: 0,
        results:
         [ { groupId:
              'consumer-group-id-672577882e482e925276-24684-d597fac0-adba-4764-957c-07cb2f4013b7',
             errorCode: 16,
             error: [KafkaJSProtocolError] } ] }   
```
Or should we maybe change the API to delete one group at a time?

Reply:
Sorry @Nevon to bother you, could you give your thought on this? I think we should definitively have an option to delete more groups at once, but is it ok to return each error as mentioned above?

Reply:
That's a good question. I'm not sure.

My gut feeling is that we should start from the assumption that deleting multiple groups  is functionally similar to deleting several within a `Promise.all`, because it could be weird to have an async function resolve with what is essentially an error. What I mean is that these two situations should behave the same:

```
// If one fails, the promise rejects
await Promise.all([a,b,c].map(deleteGroup))

// If one fails, the promise should reject with the failed groups
await deleteGroups([a,b,c])
```

So my suggestion would be that we create a new `KafkaJSDeleteGroupsError` that wraps the protocol errors and throw that if one or more groups fails to be deleted.

Something like:

```
class KafkaJSDeleteGroupsError extends KafkaJSError {
  constructor(e, { groups }) {
    super(e)

    // This is pseudo code just to show what I mean
    this.groups = {
      [groupId]: KafkaJSProtocolError
    }
  }
}
```

Reply:
Ok, will do it like that

Reply:
Ok, solved it like that, and added additional tests for this